src/amo/components/VPNPromoBanner/index.js

```diff
@@ -66,17 +74,28 @@ export const getImpressionCount = (
   return parsedCount;
 };
 
-export class VPNPromoBannerBase extends React.Component<InternalProps> {
+type State = {|
+  dismissed: boolean,
+|};
+
+export class VPNPromoBannerBase extends React.Component<InternalProps, State> {
   static defaultProps: DeafultProps = {
     _tracking: tracking,
     _localStorage: typeof window !== 'undefined' ? window.localStorage : {},
   };
 
+  constructor(props: InternalProps) {
+    super(props);
+
+    this.state = { dismissed: props.variant !== VARIANT_SHOW };
```

Comment:
nit: I don't think we specifically need to use `props.variant` here because it is also covered in `shouldShowBanner()` but I guess it's fine.

Reply:
Yeah, I started with just defaulting it to `false` in the constructor, but I figured we'd might as well default it to the correct value if `variant` isn't `VARIANT_SHOW`.